src/subdomains/core/history/mappers/transaction-dto.mapper.ts

```diff
@@ -13,7 +13,8 @@ export class TransactionDtoMapper {
       inputAmount: buyCrypto.inputAmount,
       inputAsset: buyCrypto.inputAsset,
       inputBlockchain: buyCrypto.cryptoInput?.asset.blockchain,
-      exchangeRate: buyCrypto.outputAmount ? buyCrypto.exchangeRateString : null,
+      exchangeRate: buyCrypto.outputAmount ? buyCrypto.exchangeRate.exchangeRate : null,
+      rate: buyCrypto.outputAmount ? buyCrypto.exchangeRate.rate : null,
```

Comment:
Couldn't you use `...(buyCrypto.outputAmount ? buyCrypto. buyCrypto.exchangeRate : null),` (same below)?